Set permissions for GitHub actions

naveensrinivasan opened the pull request:
<!-- NOTE: All pull requests should have tests (rspec in Ruby, qunit in JavaScript). If your code does not include test coverage, please include an explanation of why it was omitted. -->
Included permissions for the action. https://github.com/ossf/scorecard/blob/main/docs/checks.md#token-permissions
https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs

Signed-off-by: naveensrinivasan 172697+naveensrinivasan@users.noreply.github.com

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/discourse/discourse?pullRequest=16373) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/discourse/discourse?pullRequest=16373) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/discourse/discourse?pullRequest=16373) it.</sub>

pmusaraj commented:
Any feedback here @CvX ?

CvX commented:
Thanks for the PR Naveen! We're reworking workflows, and some of them will need some write permissions but read-only is a good baseline anyway. I merged the latest `main`, fixed the conflict, and added the read setting to the new workflow: #17733

naveensrinivasan commented:
> Thanks for the PR Naveen! We're reworking workflows, and some of them will need some write permissions but read-only is a good baseline anyway. I merged the latest `main`, fixed the conflict, and added the read setting to the new workflow: #17733

Thanks
